src/main/scala/Column.scala

```diff
@@ -1,22 +1,292 @@
+//      .orElse(
+//        titles.map(t => {
+//          val langNode = t.path(lang)
+//          if (langNode.isMissingNode) {
+//            langNode.elements.asScalaArray(0).asText()
+//          }
+//        })
+//      )
+
+    if (name.isDefined) {
+      Some(name.get.asInstanceOf[TextNode].asText())
+    } else if (titles.isDefined && titles.get.path(lang).isMissingNode) {
+      val langArray = Array.from(
+        titles.get.path(lang).elements().asScala
+      )
+      if (langArray.nonEmpty) {
+        Some(langArray(0).asText())
+      } else None
+    } else None // Not sure what to return here. Hope it does not reach here
+  }
+
+  def getNullParam(
+      inheritedProperties: Map[String, JsonNode]
+  ): Array[String] = {
+    inheritedProperties.get("null") match {
+      case Some(value) => {
+        value match {
+          case a: ArrayNode => {
+            var nullParamsToReturn = Array[String]()
+            val nullParams = Array.from(a.elements.asScala)
+            for (np <- nullParams)
+              nullParamsToReturn :+= np.asText()
+            nullParamsToReturn
+          }
+          case s: TextNode => Array[String](s.asText())
+        }
+      }
+      case None => Array[String]("")
+    }
+  }
+
+  def getAboutUrl(
+      inheritedProperties: Map[String, JsonNode]
+  ): Option[String] = {
+    val aboutUrlNode = inheritedProperties.get("aboutUrl")
+    aboutUrlNode match {
+      case Some(value) => Some(value.asText())
+      case _           => None
+    }
+  }
+
+  def getPropertyUrl(
+      inheritedProperties: Map[String, JsonNode]
+  ): Option[String] = {
+    val propertyUrlNode = inheritedProperties.get("propertyUrl")
+    propertyUrlNode match {
+      case Some(value) => Some(value.asText())
+      case _           => None
+    }
+  }
+
+  def getValueUrl(
+      inheritedProperties: Map[String, JsonNode]
+  ): Option[String] = {
+    val valueUrlNode = inheritedProperties.get("valueUrl")
+    valueUrlNode match {
+      case Some(value) => Some(value.asText())
+      case _           => None
+    }
+  }
+
+  def getSeparator(
+      inheritedProperties: Map[String, JsonNode]
+  ): Option[String] = {
+    val separatorNode = inheritedProperties.get("separator")
+    separatorNode match {
+      case Some(value) => Some(value.asText())
+      case _           => None
+    }
+  }
+
+  def getTitles(columnProperties: Map[String, JsonNode]): Option[JsonNode] = {
+    columnProperties.get("titles")
+  }
+
+  def fromJson(
+      number: Int,
+      columnDesc: ObjectNode,
+      baseUrl: String,
+      lang: String,
+      inheritedProperties: Map[String, JsonNode]
+  ): Column = {
+    var annotations = Map[String, JsonNode]()
+    var warnings = Array[ErrorMessage]()
+    val columnProperties = Map[String, JsonNode]()
+    val inheritedPropertiesCopy =
+      MapHelpers.deepCloneJsonPropertiesMap(inheritedProperties)
+
+    for ((property, value) <- columnDesc.getKeysAndValues) {
```

Comment:
I'd refactor this into a function similar to the ones we've seen in `TableGroup` and `Table` which partition the properties based on their type.